cmake/GtsamBuildTypes.cmake

```diff
@@ -183,18 +183,33 @@ set(CMAKE_EXE_LINKER_FLAGS_PROFILING ${GTSAM_CMAKE_EXE_LINKER_FLAGS_PROFILING})
 
 # Clang uses a template depth that is less than standard and is too small
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-    # Apple Clang before 5.0 does not support -ftemplate-depth.
-    if(NOT (APPLE AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "5.0"))
-        list_append_cache(GTSAM_COMPILE_OPTIONS_PUBLIC "-ftemplate-depth=1024")
-    endif()
+  # Apple Clang before 5.0 does not support -ftemplate-depth.
+  if(NOT (APPLE AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "5.0"))
+    list_append_cache(GTSAM_COMPILE_OPTIONS_PUBLIC "-ftemplate-depth=1024")
+  endif()
 endif()
 
 if (NOT MSVC)
-  option(GTSAM_BUILD_WITH_MARCH_NATIVE  "Enable/Disable building with all instructions supported by native architecture (binary may not be portable!)" OFF)
-  if(GTSAM_BUILD_WITH_MARCH_NATIVE AND (APPLE AND NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64"))
-    # Add as public flag so all dependant projects also use it, as required
-    # by Eigen to avid crashes due to SIMD vectorization:
-    list_append_cache(GTSAM_COMPILE_OPTIONS_PUBLIC "-march=native")
+  # Add as public flag so all dependant projects also use it, as required
+  # by Eigen to avid crashes due to SIMD vectorization:  option(GTSAM_BUILD_WITH_MARCH_NATIVE  "Enable/Disable building with all instructions supported by native architecture (binary may not be portable!)" OFF)
+  if(APPLE AND (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang") AND ("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "15.0"))
+    if(GTSAM_BUILD_WITH_MARCH_NATIVE)
```

Comment:
Can you please move this check to the outside since there is no `else` condition?

Something like the below is equivalent I believe

```cmake
if(GTSAM_BUILD_MARCH_NATIVE)
    if(APPLE AND (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang") AND ("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "15.0"))

        if(NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
            list_append_cache(GTSAM_COMPILE_OPTIONS_PUBLIC "-march=native")
        else()
            message(WARNING "The option GTSAM_BUILD_WITH_MARCH_NATIVE is ignored, because native architecture is not supported.")
        endif() # CMAKE_SYSTEM_PROCESSOR

    else()
        include(CheckCXXCompilerFlag)
        CHECK_CXX_COMPILER_FLAG("-march=native" COMPILER_SUPPORTS_MARCH_NATIVE)
        if(COMPILER_SUPPORTS_MARCH_NATIVE)
            list_append_cache(GTSAM_COMPILE_OPTIONS_PUBLIC "-march=native")
        else()
            message(WARNING "The option GTSAM_BUILD_WITH_MARCH_NATIVE is ignored, because native architecture is not supported.")
        endif()  # COMPILER_SUPPORTS_MARCH_NATIVE

    endif()
endif()  # GTSAM_BUILD_WITH_MARCH_NATIVE
```

Plus add some formatting, since this is a bit hard to parse?
Or preferably, use a [generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html)?

Reply:
PS please correct me if my rearrangement of the if-blocks is wrong. I prefer generator expressions for this reason. 😅

Reply:
Indeed, Varun, your order makes much more sense. Sorry, I am not a cmake expert and do not master generator expressions. Thus, rewrote it with if-blocks. --- Again, I could not test it on Apple platform because of lack of hardware. Tested only with Ubuntu on Intel machine.